plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConfig.java

```diff
@@ -131,9 +78,12 @@ public OptionalInt getParallelism()
 
     @Config("bigquery.parallelism")
     @ConfigDescription("The number of partitions to split the data into.")
-    public BigQueryConfig setParallelism(int parallelism)
+    public BigQueryConfig setParallelism(Integer parallelism)
```

Comment:
Things read from config files are never null IIRC. Revert the nullability related changes.

Reply:
Ah, we weren't using `assertRecordedDefaults` here earlier. Can you extract a prep commit if not too painful. Would make it easier to understand what happenned.

Looks good otherwise.